glean_parser/util.py

```diff
@@ -120,15 +150,20 @@ def load_yaml_or_json(path: Path, ordered_dict: bool = False):
     if TESTING_MODE and isinstance(path, dict):
         return path
 
+    content = {}
     if path.suffix == ".json":
         with path.open("r", encoding="utf-8") as fd:
-            return json.load(fd)
+            content = json.load(fd)
     elif path.suffix in (".yml", ".yaml", ".yamlx"):
         with path.open("r", encoding="utf-8") as fd:
             if ordered_dict:
-                return ordered_yaml_load(fd)
+                content = ordered_yaml_load(fd)
             else:
-                return yaml.load(fd, Loader=_NoDatesSafeLoader)
+                content = yaml.load(fd, Loader=SafeLineLoader)
+        if add_line_number:
+            return content
+        else:
+            return remove_output_params(content, "defined_in")
```

Comment:
`add_line_number` param was added here to clarify when [we need to add line number to the output](https://github.com/Iinh/glean_parser/blob/0e95f993a4a715c467c405f6b673b04532058b51/glean_parser/parser.py#L64-L67) versus [when we don't](https://github.com/Iinh/glean_parser/blob/0e95f993a4a715c467c405f6b673b04532058b51/glean_parser/parser.py#L109).